testlib/s3mock.py

```diff
@@ -202,58 +166,16 @@ def ln(part):
     return b"\n".join(output)
 
 
-class RecordingAdapterShim:
-    """Adapter wrapper for recording requests and responses
-
-    Usage::
-
-        # This is the original adapter
-        adapter = get_adapter()
-
-        # Generate the shim, wrap the adapter and set the
-        # log filename.
-        recording_adapter = RecordingAdapterShim()
-        recording_adapter.wrapped_adapter = adapter
-        recording_adapter.filename = 's3mock.log'
-
-        # Now you have a wrapped adapter that will record HTTP
-        # conversations.
-        adapter = recording_adapter
-
-    """
-
-    def send(self, request, *args, **kwargs):
-        with open(self.filename, "ab") as fp:
-            fp.write(b"===================================\n")
-            fp.write(b"REQUEST>>>\n")
-            fp.write(serialize_request(request))
-            fp.write(b"-----\n")
-
-        response = self.wrapped_adapter.send(request, *args, **kwargs)
-
-        with open(self.filename, "ab") as fp:
-            fp.write(b"RESPONSE<<<\n")
-            fp.write(serialize_response(response))
-
-        return response
-
-    def __getattr__(self, name):
-        return getattr(self.wrapped_adapter, name)
-
-
 class S3Mock:
     """Provide a configurable mock for Boto3's S3 bits.
 
-    Boto3 uses botocore which uses s3transfer which uses requests to do REST
-    API calls.
+    Boto3 uses botocore which uses s3transfer which uses urllib3 to do REST API calls.
 
-    ``S3Mock`` mocks requests by creating a fake adapter which allows it to
-    intercept all outgoing HTTP requests. This lets us do two things:
```

Comment:
Hilariously, I said "two things" here which used to be wrong, but now it's right!